### DISCUSSION TITLE
Duplicate active alerts returned from API?

### DISCUSSION THREAD

**Discussion by aeharding** [weather-gov/api#573] [Q&A]
Hello!

I am currently querying active alerts for the PDX airport. Via the website I see one heat advisory - https://forecast.weather.gov/MapClick.php?lat=45.5887&lon=-122.5969 but when I query the API  (GET https://api.weather.gov/alerts/active?point=45.509,-122.649&message_type=alert&status=actual) I see 3 heat advisories returned, with different issued dates, for the same time period.

Full response: [response.json.txt](https://github.com/weather-gov/api/files/9277566/response.json.txt)

Reference headers:
```json
{
  "name": "x-request-id",
  "value": "eac56848-474f-4f94-b786-11105a46729e"
},
{
  "name": "x-server-id",
  "value": "vm-lnx-nids-apiapp8.ncep.noaa.gov"
}
```

Am I querying the API incorrectly? If not, is it safe to dedupe by product ID, and use the latest issued one?

Thanks! This API is truly exceptional, and I've had very few issues with it overall!
Alex

**Comment by scadergit** [comment]
This is a known issue. The upstream source of alerts for the API changed the parameter for alert event chains (a new parameter named expiredReferences), and the API has not been updated. This breaks the ability for the API to reconstruct the event chain resulting in expired alerts showing on the active endpoint. You can use the expiredReferences parameter to resolve the chain manually.

**Reply by aeharding** [reply]
Thanks! I've hacked the following together until the API is updated.

```ts
  data.features = data.features.filter(
    (feauture) =>
      !data.features.find((potentialNewerFeature) =>
        potentialNewerFeature.properties.parameters.expiredReferences?.find(
          (reference) => reference.includes(feauture.properties.id)
        )
      )
  );
```

**Reply by aeharding** [reply]
Hi @scadergit, I no longer see the `expiredReferences` property. I'm also seeing duplicate alerts. Any idea for a workaround and/or when this will be resolved on the API side of things? Thanks so much and have a great weekend

**Reply by aeharding** [reply]
Update: Now using

```ts
  data.features = data.features.filter(
    (feature) =>
      !data.features.find(
        (potentialNewerFeature) =>
          potentialNewerFeature.properties.event === feature.properties.event &&
          Date.parse(potentialNewerFeature.properties.sent) >
            Date.parse(feature.properties.sent)
      )
  );
```

Would still appreciate a fix from the API side of things since this does NOT cover all edge cases. :)

**Reply by estebaninformatica** [reply]
@aeharding 

If you have two or more events in the same area, you could lose events.

I am right ?

this solution seems more correct to me:

```javascript
  data.features = data.features.filter(
    (feauture) =>
      !data.features.find((potentialNewerFeature) =>
        potentialNewerFeature.properties.parameters.expiredReferences?.find(
          (reference) => reference.includes(feauture.properties.id)
        )
      )
  );
```

@scadergit 

Is there any update on this?

Do you see possible to add a group_id to the api?

Thank you

**Reply by aeharding** [reply]
I saw some dupes without `expiredReferences` unfortunately. Would also appreciate an update.
